livre/lib/cache/TextureCache.cpp

```diff
@@ -29,45 +30,69 @@
 
 namespace livre
 {
-
-TextureCache::TextureCache( const GLint internalTextureFormat )
-    : texturePoolFactory_( internalTextureFormat )
+struct TextureCache::Impl
 {
-    statisticsPtr_->setStatisticsName( "Texture cache GPU");
-}
+    Impl( TextureDataCache& dataCache,
+          const int internalTextureFormat )
+        : _dataCache( dataCache )
+    {
+        const VolumeDataSource& dataSource = _dataCache.getDataSource();
+        const VolumeInformation& info = dataSource.getVolumeInformation();
+
+        uint32_t format;
+        switch( info.compCount )
+        {
+            case 1:
+                format = GL_RED;
+                break;
+            case 3:
+                format = GL_RGB;
+                break;
+            default:
+                format = GL_RED;
+                break;
+        }
```

Comment:
The exact same switch() is above. It also seems unrobust, handling two cases only with no throw/error reporting